templates/main.ejs

```diff
@@ -37,21 +37,33 @@
             const theme = getCookie('data-theme') === 'dark' ? 'light' : 'dark';
             setTheme(theme);
         }
+
+        function navigateTo(url) {
+            window.location.href = url;
+        }
+    </script>
+    <script>
+      function setResponseCookies(response) {
```

Comment:
I was experimenting around with something response-related. 
To make the code more manageable, I put the code into a separate script. 
Feel free to revert it to EJS later, I do not feel too strongly about this. But let's not block the PR for this.